Sources/NIOCore/EventLoopFuture.swift

```diff
@@ -163,7 +163,7 @@ public struct EventLoopPromise<Value> {
     ///     - file: The file this promise was allocated in, for debugging purposes.
     ///     - line: The line this promise was allocated on, for debugging purposes.
     @inlinable
-    internal init(eventLoop: EventLoop, file: StaticString, line: UInt) {
+    internal init(eventLoop: EventLoop, file: StaticString = "cannot leak or bug in NIO", line: UInt = #line) {
```

Comment:
I assume you set this string, so that you don't always need to pass the same string, when creating a Promise. I assume it is only used in `makePromise` from here on. I think we should add a code comment here, explaining the string and when it is important to overwrite.

Reply:
@fabianfett PTAL, I made the code cleaner I think. @Lukasa too